### PR TITLE
refactor configuration.

### PR DESCRIPTION
**Pull request by osheroff**
I'm prepping to have multiple maxwells running inside the same JVM.

This PR is best viewed side-by-side if you ask me; too many unrelated changes in the same place.

@zendesk/rules

### CODE REVIEW

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 1]
```diff
@@ -124,52 +123,71 @@ private String parseLogLevel(String level) {
...
+			return (String) options.valueOf(name);
+		else if ( (properties != null) && properties.containsKey(name) )
+			return (String) properties.getProperty(name);
+		else
+			return defaultVal;
+	}
+
+	private MaxwellMysqlConfig parseMysqlConfig(String prefix, OptionSet options, Properties properties) {
+		MaxwellMysqlConfig config = new MaxwellMysqlConfig();
+		config.host     = fetchOption(prefix + "host", options, properties, null);
```
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: default *is* actually localhost, but we spit out a warning down in `validate`

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java [hunk 2]
```diff
@@ -124,52 +123,74 @@ private String parseLogLevel(String level) {
...
 
 		if ( options.has("help") )
 			usage("Help for Maxwell:");
 
-		if ( options.has("log_level")) {
-			this.log_level = parseLogLevel((String) options.valueOf("log_level"));
-		}
-
-		this.maxwellMysql.parseOptions("", options);
+		this.log_level = fetchOption("log_level", options, properties, null);
```
Comment (dadah89) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: wondering if setting `properties` and `options` to instance variables (and unsetting them before leaving `parse`) so they won't be repeated with all the method calls wouldn't look better

**Comment by dadah89** [comment]
lgtm 👍

File: src/main/java/com/zendesk/maxwell/MaxwellConfig.java (hunk 2 above)
Comment (osheroff) [inline src/main/java/com/zendesk/maxwell/MaxwellConfig.java]: could go down that road, but the right there would be to be like:

```
  new MaxwellConfigOptionParser(this, options, properties).parse()
```

or something.  Leaving instance vars around just to not pass them into other methods (ie they'll get thrown away at the end of the method) is IMO kinda bad juju.

**Pull request merged**